File: implementations/csharp/ILlmInvocation.cs
namespace Interfaces
{
    /// <summary>
    /// Contract for the main invocation object
    /// </summary>
    public interface ILlmInvocation
    {
        Models.ConversationState Conversation {  get; }
        Models.RagContext Rag { get; }
        Models.MemoryContext Memory { get; }
        Models.ToolContext Tools { get; }
        Models.GenerationOptions Options { get; }
        Models.ExecutionHints Hints { get; }

        /// <summary>
        /// Check internal coherency before the run
        /// </summary>
        IvalidationResult Validate();
    }
}

File: implementations/vb/Loom.Core/Interfaces/ILlmInvocation.vb
Namespace Interfaces
    ''' <summary>
    ''' Contract for the main invocation object
    ''' </summary>
    Public Interface ILlmInvocation
        ReadOnly Property Conversation As Models.ConversationState
        ReadOnly Property Rag As Models.RagContext
        ReadOnly Property Memory As Models.MemoryContext
        ReadOnly Property Tools As Models.ToolContext
        ReadOnly Property Options As Models.GenerationOptions
        ReadOnly Property Hints As Models.ExecutionHints

        ''' <summary>
        ''' Check internal coherency before the run
        ''' </summary>
        Function Validate() As IValidationResult

    End Interface
End Namespace